executor/src/subscriber.rs

```diff
@@ -120,144 +153,155 @@ impl Subscriber {
+    async fn fetch_payload(
+        &self,
+        digest: BatchDigest,
+        worker_id: WorkerId,
+        authorities: Vec<PublicKey>,
+    ) -> Batch {
+        if let Some(payload) = self.try_fetch_locally(digest, worker_id).await {
+            return payload;
+        }
+        let _timer = self.metrics.subscriber_remote_fetch_latency.start_timer();
+        let mut stagger = Duration::from_secs(0);
+        let mut futures = vec![];
+        for authority in &authorities {
+            let worker = self.worker_cache.load().worker(authority, &worker_id);
+            let worker = match worker {
+                Ok(worker) => worker.name,
+                Err(err) => {
+                    error!(
+                        "Worker {} not found for authority {}: {:?}",
+                        worker_id, authority, err
+                    );
+                    continue;
                 }
-            }
-            Err(err) => {
-                panic!("Irrecoverable error occurred while retrieving block payload: {err}");
-            }
+            };
+            let future = self.fetch_from_worker(stagger, worker, digest, authority.clone());
+            futures.push(future.boxed());
+            stagger += Duration::from_secs(1);
         }
-
-        Ok(())
+        let (batch, _, _) = futures::future::select_all(futures).await;
+        batch
     }
 
-    /// The wait_on_payload will try to retrieve the certificate's payload
-    /// from the workers via the block_waiter component and relase the
-    /// `deliver` once successfully done. Since we want the output to be
-    /// sequenced we will not quit this method until we have successfully
-    /// fetched the payload.
-    async fn wait_on_payload(
-        metrics: Arc<ExecutorMetrics>,
-        back_off_policy: ExponentialBackoff,
-        store: Store<(CertificateDigest, BatchDigest), Batch>,
-        tx_get_block_commands: metered_channel::Sender<BlockCommand>,
-        deliver: ConsensusOutput,
-    ) -> SubscriberResult<ConsensusOutput> {
-        // the latency will be measured automatically once the guard
-        // goes out of scope and dropped
-        let _start_guard = metrics.subscriber_download_payload_latency.start_timer();
-        let mut attempts_count = 0;
-
-        let get_block = move || {
-            let message = deliver.clone();
-            let certificate_id = message.certificate.digest();
-            let tx_get_block = tx_get_block_commands.clone();
-            let batch_store = store.clone();
-            let executor_metrics = metrics.clone();
-            let attempts = {
-                attempts_count += 1;
-                attempts_count
-            };
-
-            let span = debug_span!("get_block", attempt = attempts);
-            async move {
-                let (sender, receiver) = oneshot::channel();
-
-                tx_get_block
-                    .send(BlockCommand::GetBlock {
```

Comment:
I think this was the only usage of `BlockCommand::GetBlock`. We can add a comment above `BlockCommand::GetBlock` to mention that it is now unused and see if we can delete unnecessary code.

Reply:
I will take a look if block waiter can be removed after this PR (also will need to sync with @akichidis about it)